react/features/lobby/actions.js

```diff
@@ -0,0 +1,176 @@
+ *
+ * @returns {Function}
+ */
+export function cancelKnocking() {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        if (typeof APP !== 'undefined') {
+            // when we are redirecting the library should handle any
+            // unload and clean of the connection.
+            APP.API.notifyReadyToClose();
+            dispatch(maybeRedirectToWelcomePage());
+
+            return;
+        }
+
+        dispatch(conferenceLeft(getCurrentConference(getState)));
+        dispatch(appNavigate(undefined));
+    };
+}
+
+/**
+ * Action to be dispatched when a knocking poarticipant leaves before any response.
+ *
+ * @param {string} id - The ID of the participant.
+ * @returns {hideNotification}
+ */
+export function knockingParticipantLeft(id: string) {
+    return hideNotification(hashCode(id));
+}
+
+/**
+ * Action to set the knocking state of the participant.
+ *
+ * @param {boolean} knocking - The new state.
+ * @returns {{
+ *     state: boolean,
+ *     type: SET_KNOCKING_STATE
+ * }}
+ */
+export function setKnockingState(knocking: boolean) {
+    return {
+        knocking,
+        type: SET_KNOCKING_STATE
+    };
+}
+
+/**
+ * Starts knocking and waiting for approval.
+ *
+ * @param {string} password - The password to bypass knocking, if any.
+ * @returns {Function}
+ */
+export function startKnocking(password?: string) {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const { membersOnly } = getState()['features/base/conference'];
+        const localParticipant = getLocalParticipant(getState());
+
+        dispatch(setKnockingState(true));
+        dispatch(conferenceWillJoin(membersOnly));
+        membersOnly
+            && membersOnly.joinLobby(localParticipant.name, localParticipant.email, password ? password : undefined);
+    };
+}
+
+/**
+ * Action to open the lobby screen.
+ *
+ * @returns {openDialog}
+ */
+export function openLobbyScreen() {
+    return openDialog(LobbyScreen);
+}
+
+/**
+ * Action to set the new state of the lobby mode.
+ *
+ * @param {boolean} enabled - The new state to set.
+ * @param {string} password - The password to set if enabled.
+ * @returns {{
+ *     enabled: boolean,
+ *     type: SET_LOBBY_MODE_ENABLED
+ * }}
+ */
+export function setLobbyModeEnabled(enabled: boolean, password?: string) {
+    return {
+        enabled,
+        password: enabled && password ? password : undefined,
+        type: SET_LOBBY_MODE_ENABLED
+    };
+}
+
+/**
+ * Action to be executed when a participant starts knocking into the conference.
+ *
+ * @param {string} id - The id of the knocking participant.
+ * @param {string} name - The name of the knocking participant.
+ * @param {string} avatarInfo - Any info that helps us render an avatar (avatarID, URL...etc).
+ * @returns {Function}
+ */
+export function participantIsKnocking(id: string, name: string, avatarInfo: string) {
+    logger.info(`Participant is knocking. Id: ${id}, name: ${name}`);
```

Comment:
logger.debug?

Reply:
it's a good question :) i think it's important to see this in the logs even in callstats later, and i think debug logs don't go to callstats. Amirite?

Reply:
All of them go, depends on the configured level, which by default is trace.